### PR TITLE
setup postgres database connection with node js ecommerce app

### PR DESCRIPTION
**Pull request by BenoitIT**
### What does this PR do?
This has functions and methods to set up connection  [ecommerce-app-legends-bn](https://github.com/atlp-rwanda/ecommerce-app-legends-bn) App with postgres database during development phase.

#### what are the nature of PL contents?
---config folder which has js file with db connection configurations
---models folder which has User model created for functionality testing as well index.js which has db connection with 
     models
---migration folder which store table structures to be migrated to the database
---scripts to run both create migration events as well as make: db migrate
---connection folder which has folder to notify the the status of connection from the node js console

N.B : migrations and models folder are still in progress as new and more migrations and modls will b created

### Description of Tasks completed
----database connection set-up successfully
----first model created successfully
----first migration created and migrated successfully
--- make model and create migration as well as migrate scripts added and tested
### how to test this?
clone repo , branch git clone https://github.com/atlp-rwanda/ecommerce-app-legends-bn.git
-`run npm i`
-have `postgres database` installed 
-create a database with `ecommerce_db`as its name
-create .env file
-under .env file initialize:`DEV_DATABASE_URL=postgres://<username>:<password>@127.0.0.1:5432/ecommerce_db`
###### remember to take away angled bracket  beside username and password as well colon between those two from connection string above
-start the app and check connectivity in development `npm run dev `
-  run `npm run make:migration`  in terminal to make migation  to create new migration
-rename created migration file with your desired file
- run`npm run db:migrate` to create new table in database
-  run`npm run generate:seed` to create new database seed 
- run`npm run db:seed` to seed the database
-  run`npm run migrate:undo` to revert migrations
Screen Shot 2023-03-02 at 10 57 22
 database seed

### CODE REVIEW

**Comment by mansurissa** [comment]
Put `database` folder in `src`

File: package.json [hunk 1]
```diff
@@ -7,6 +7,11 @@
     "build": "babel src -d dist",
     "start": "nodemon dist/index.js",
     "dev": "nodemon --exec babel-node src/index.js",
+    "migrate": "npx sequelize-cli db:migrate",
```
Comment (mansurissa) [inline package.json]: I would suggest before migrating and seeding to first undo migrations and seeds

File: src/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import express from 'express';
 import rootRouter from './api/root';
 import docs from './docs/index';
+import { testConnection } from '../database/models';
```
Comment (mansurissa) [inline src/index.js]: `testConnection` sounds like you are testing something. a name like `databaseConnection` would be better

**Review by mansurissa — CHANGES_REQUESTED** [review]
Work on the comments, requested changes and ignore `.DS_store`

**Review by Aimeana100 — CHANGES_REQUESTED** [review]
Thank you  @BenoitIT, you have made really good job. 
 I reviwed your work, and i foud that you may have been sone some `mistakes` you have made.
- ` newschema--attributes ` in  ` "make:migration":"npx sequelize-cli model:generate --name newschema--attributes firstName:string,lastName:string,email:string",` should be space separate
- * Again  update your pull request message* approprietry `ex. ` : `npm run db:migrate` doesn't make sence.
-  Lastly i want to suggest that you follow the file structure:
--   Letting Database be in src folder shold be much better.
--  even if it is sequelize-cli template i would like to advice you, to also take `models` folder direct under src *i.e* : `src/models` and update `.eslintrc.js`  accordingly.

**Comment by Aimeana100** [comment]
> * npm run db:seed


> Work on the comments, requested changes and ignore `.DS_store`

and @BenoitIT  may ignore `package-lock.json` as someone has suggested previously.

**Comment by BenoitIT** [comment]
thanks for feedback every mentioned response is on the way getting handled!

**Pull request closed without merging**